tests/test_scopes.py

```diff
@@ -482,3 +482,41 @@ def test_tekscopesw(device_manager: DeviceManager) -> None:
     assert scope.ip_address == ""
     assert not scope.total_channels
     assert scope.all_channel_names_list == ()
+
+
+def test_tekscope2k(device_manager: DeviceManager) -> None:
+    """Test the tekscope2k implementation.
+
+    Args:
+        device_manager: The DeviceManager object.
+    """
+    device_manager.remove_all_devices()
+    # Verify hostname can be determined when only the IP is provided
+    scope: MSO2KB = device_manager.add_scope(
+        "MSO2KB-SERIAL1", alias="mso2kb", connection_type="USB"
+    )
+
+    assert scope.idn_string == "TEKTRONIX,MSO2024B,001061,CF:91.1CT FV:v1.30"
+
+    assert scope.all_channel_names_list == ("CH1", "CH2", "CH3", "CH4")
+    assert scope.total_channels == 4
+
+    # Test toggle all channels on/off
+    scope.turn_all_channels_on()
+    scope.turn_all_channels_off()
```

Comment:
There should be some kind of check here that the channels are off/on.

Reply:
Probably something like this:
```python
    for ch_name in scope.all_channel_names_list:
        assert scope.query(f"SELECT:{ch_name}?") == "0"  # or "1" depending on which method you are validating
```

Reply:
Tests added for turn_all_channels_on/off.